src/CsvWriter.php

```diff
@@ -0,0 +1,150 @@
+<?php
+
+namespace Keboola\Csv;
+
+class CsvWriter extends AbstractCsvFile
+{
+    /**
+     * @var resource
+     */
+    protected $filePointer;
+
+    /**
+     * @var string
+     */
+    private $mode;
+
+    /**
+     * CsvFile constructor.
+     * @param string $fileName
+     * @param string $delimiter
+     * @param string $enclosure
+     * @param string $mode
+     * @throws InvalidArgumentException
+     */
+    public function __construct(
+        $fileName,
+        $delimiter = self::DEFAULT_DELIMITER,
+        $enclosure = self::DEFAULT_ENCLOSURE,
+        $mode = 'w'
+    ) {
+        parent::__construct($fileName);
+        $this->setMode($mode);
+        $this->setDelimiter($delimiter);
+        $this->setEnclosure($enclosure);
+        $this->openCsvFile();
+    }
+
+    public function __destruct()
+    {
+        $this->closeFile();
+    }
+
+    /**
+     * @param array $row
+     * @throws Exception
+     */
+    public function writeRow(array $row)
+    {
+        $str = $this->rowToStr($row);
+        $ret = fwrite($this->getFilePointer(), $str);
+
+        /* According to http://php.net/fwrite the fwrite() function
+         should return false on error. However not writing the full
+         string (which may occur e.g. when disk is full) is not considered
+         as an error. Therefore both conditions are necessary. */
+        if (($ret === false) || (($ret === 0) && (strlen($str) > 0))) {
+            throw new Exception(
+                "Cannot write to file {$this->getPathname()}",
+                Exception::WRITE_ERROR,
+                null,
+                Exception::WRITE_ERROR_STR
+            );
+        }
+    }
+
+    /**
+     * @param array $row
+     * @return string
+     * @throws Exception
+     */
+    public function rowToStr(array $row)
+    {
+        $return = [];
+        foreach ($row as $column) {
+            if (!(
+                is_scalar($column)
+                || is_null($column)
+                || (
+                    is_object($column)
+                    && method_exists($column, '__toString')
+                )
+            )) {
+                $type = gettype($column);
+                throw new Exception(
+                    "Cannot write {$type} into a column",
+                    Exception::WRITE_ERROR,
+                    null,
+                    Exception::WRITE_ERROR_STR,
+                    ['column' => $column]
+                );
+            }
+
+            $return[] = $this->getEnclosure() .
+                str_replace($this->getEnclosure(), str_repeat($this->getEnclosure(), 2), $column) .
+                $this->getEnclosure();
+        }
+        return implode($this->getDelimiter(), $return) . "\n";
+    }
+
+    /**
+     * @return resource
+     */
+    protected function getFilePointer()
```

Comment:
Ta metoda je stejná pro reader i writer, ne? Tak by mohla být v tom abstraktním předkovi.

Reply:
je, a puvodne jsem ji tam mel, ale je to na palici, protoze ten pointer ma jinou inicializaci, takze zusatane v abstraktni tride nesmyslnej fragment, a vlastne je teda metoda stejna jen "shodou okolnosti"

Reply:
tak nakonec jsem to udelal :) pridanim tech pointeru tam vzniklo vic spolecnyho kodu a zda se , ze to vyresila jedna abstraktni metoda